### DISCUSSION TITLE
Is there a way to use the MNIST example *without* having it trying to download Python & fill out a venv?

### DISCUSSION THREAD

**Discussion by crabdancing** [tracel-ai/burn#1470] [General]
I am using the ndarray backend, which should be pure Rust, no?

The command is: `cargo run --example mnist --release --features ndarray`

**Comment by laggui** [comment]
Hmmm this should have been fixed, we changed our MNIST dataset to use the raw data instead of pulling from HuggingFace datasets (which required a python env) in #1176.

Are you seeing an error?

**Reply by crabdancing** [reply]
Not seeing an error of any sort. Also tried RUST_LOG=debug for good measure (err, not sure what other env var it might be listening to, but it didn't seem to change anything).

**Reply by laggui** [reply]
What version of Burn are you using? The MNIST example doesn't set up a venv like some other examples.

**Reply by crabdancing** [reply]
Ooof. Looks like I was on 58273a84411c77dd2a4b81d415c7c4dfc7d686b9, but that actually looks p. old. I'll try with c729401fb268ae0451ed2695f1b31653fb9207de.

**Reply by crabdancing** [reply]
Looks like it works now! Thanks for pointing out my obvious oversight! :D

**Comment by laggui** [comment]
I just realized though we forgot to change a paragraph in the book with that change.

> Typically, one trains a model on some dataset. Burn provides a library of very useful dataset sources and transformations. In particular, there are Hugging Face dataset utilities that allow to download and store data from Hugging Face into an SQLite database for extremely efficient data streaming and storage. **For this guide, we will use the MNIST dataset provided by Hugging Face.**

The bolded part is not true anymore. Will fix.

/edit: see PR #1471

**Comment by crabdancing** [comment]
Resolved by update. Glad I could point out the documentation thing. Updating to latest commit fixes issue.
